stl/inc/generator

```diff
@@ -9,9 +9,9 @@
 #include <yvals_core.h>
 #if _STL_COMPILER_PREPROCESSOR
 
-#if !_HAS_CXX23 || !defined(__cpp_impl_coroutine) || !defined(__cpp_lib_concepts) // TRANSITION, GH-395
-_EMIT_STL_WARNING(STL4038, "The contents of <generator> are available only with C++23 or later coroutine support.")
-#else // ^^^ no coroutine support / coroutines vvv
+#if !_HAS_CXX23
+_EMIT_STL_WARNING(STL4038, "The contents of <generator> are available only with C++23 or later.");

```

Comment:
I led you astray here: we need the `!defined(__cpp_impl_coroutine)` (and the vague "coroutine support" in the error message, which could probably be improved) to disable the header when `/await` is on the command line. `/await` tells MSVC to disable the standard coroutine ABI and use the pre-standard ABI instead. We do _not_ want to deal with this potential ODR fiasco, so I think we should revert the changes to lines 12, 13 and 471.

Reply:
I'm mistaken: `<coroutine>` (which `<generator>` includes) will `_EMIT_STL_ERROR` when included with `/await`, so `<generator>` need not guard against `/await`.

Reply:
It's actually `_EMIT_STL_WARNING` followed by expanding to nothing, which will presumably cause `<generator>` to catastrophically fail to compile:

https://github.com/microsoft/STL/blob/192a84008a59ac4d2e55681e1ffac73535788674/stl/inc/coroutine#L11-L14

This is just a quality of diagnostic issue, so I'll proceed with this merge and we can do something about it later if we want. Thanks for looking into this.